refactor(components): align KolEvent names with native DOM event names

deleonio opened the pull request:
## What changed?

This PR renames `KolEvent` enum values to match native DOM event names (e.g. `KolEvent.close` → `'close'`). E2E tests that passed the `KolEvent` enum directly to Playwright's `evaluate()` are updated to use string literals, fixing Playwright's inability to serialise enums. A TypeScript compilation error in `kolibri-cli` caused by a `readonly` array mismatch is also resolved using a spread-operator copy. 33 files are changed.

## Linked issues

- Refs #9126 — KolEvent native DOM event names

## Type of change

- [ ] ✨ New feature
- [x] 🔼 Improvement
- [ ] 🐛 Bug fix
- [ ] 💥 Breaking change
- [ ] 🔧 Internal

## Checklist

- [x] PR title follows Conventional Commits format (`refactor(components): align KolEvent names with native DOM event names`)

> ℹ️ Original title `Align KolEvent names with native DOM events` was corrected to conform to Conventional Commits format.

<details>
<summary>🇩🇪 Deutsche Zusammenfassung</summary>

### Was wurde geändert?

Dieses PR benennt `KolEvent`-Enum-Werte so um, dass sie nativen DOM-Event-Namen entsprechen. E2E-Tests werden auf String-Literale umgestellt, da Playwright Enums nicht serialisieren kann. Ein TypeScript-Compilerfehler in `kolibri-cli` durch ein `readonly`-Array wird ebenfalls behoben.

### Verknüpfte Tickets

- Refs #9126 — KolEvent native DOM Event-Namen

### Art der Änderung

- [x] 🔼 Verbesserung

### Geänderte Dateien

- 33 Dateien in Components, E2E-Tests und kolibri-cli

</details>

deleonio commented on packages/components/src/utils/events.ts:
```diff
@@ -1,30 +1,35 @@
-	mousedown = 'kolMousedown',
-	reset = 'kolReset',
-	select = 'kolSelect',
-	selectionChange = 'kolSelectionChange',
-	sort = 'kolSort',
-	submit = 'kolSubmit',
-	toggle = 'kolToggle',
+	changePage = 'changePage',
+	changePageSize = 'changePageSize',
+	click = 'click',
+	close = 'close',
+	create = 'create',
+	focus = 'focus',
+	input = 'input',
+	keydown = 'keydown',
+	mousedown = 'mousedown',
+	reset = 'reset',
+	select = 'select',
+	selectionChange = 'selectionChange',
+	sort = 'sort',
+	submit = 'submit',
+	toggle = 'toggle',
```
Wir wollen das aber genau so.

deleonio commented:
@codex fix ci

deleonio commented on packages/tools/kolibri-cli/test/rename-kol-event-names.spec.ts:
```diff
@@ -0,0 +1,436 @@
+			assert.ok(!content.includes('kolClose'));
+			assert.ok(!content.includes('kolSubmit'));
+		});
+
+		it('does not replace partial matches without word boundaries', () => {
+			const tsPath = path.join(tmpDir, 'event-no-boundary.ts');
+			fs.writeFileSync(
+				tsPath,
+				`const mykolChange = 'something';
+const kolChangeInMyCode = () => {};
+// This should be replaced because it has word boundaries:
+element.addEventListener('kolChange', handler);`,
+			);
+
+			const task = RenameKolEventNamesTask.getInstance('^4');
+			task.run(tmpDir);
+
+			const content = fs.readFileSync(tsPath, 'utf8');
+			// Variables and identifiers WILL be replaced due to word boundaries
+			// This documents current behavior - they are separate words
+			assert.ok(content.includes('mychange') || content.includes('mykolChange'));
+			assert.ok(content.includes('changeInMyCode') || content.includes('kolChangeInMyCode'));
+			// But the event listener should definitely be replaced
+			assert.ok(content.includes("addEventListener('change', handler)"));
+		});
```
is done

laske185 commented on docs/BREAKING_CHANGES.v4.md:
```diff
@@ -31,6 +31,46 @@ import { defineCustomElements } from '@public-ui/components/loader';
+| `kolMousedown`         | `mousedown`         |
+| `kolReset`             | `reset`             |
+| `kolSelect`            | `select`            |
+| `kolSelectionChange`   | `selectionChange`   |
```
Die nativen events sind komplett lowercase, nicht camelcase!
https://developer.mozilla.org/en-US/docs/Web/API/Document/selectionchange_event

deleonio replied:
@laske185 - Ja, dann lass uns das gleich mal alles lower machen, oder?

laske185 commented on packages/components/src/utils/events.ts:
```diff
@@ -1,30 +1,34 @@
+	mousedown = 'mousedown',
+	reset = 'reset',
+	select = 'select',
+	selectionChange = 'selectionChange',
```
Somit auch hier auf lowercase ändern.

deleonio replied:
Passt so aber.

laske185 commented on packages/tools/kolibri-cli/src/migrate/runner/tasks/v4/events.ts:
```diff
@@ -0,0 +1,115 @@
+	kolMousedown: 'mousedown',
+	kolReset: 'reset',
+	kolSelect: 'select',
+	kolSelectionChange: 'selectionChange',
```
Und hier auch auf lowercase ändern.

deleonio replied:
Passt so aber.

laske185 commented on packages/components/src/components/alert/alert.e2e.ts:
```diff
@@ -24,11 +23,11 @@ test.describe('kol-alert', () => {
+			const eventPromise = page.locator('kol-alert').evaluate(async (element: HTMLKolAlertElement) => {
 				return new Promise((resolve) => {
-					element.addEventListener(KolEvent.close, resolve);
+					element.addEventListener('close', resolve);
```
Warum das Replacement und nicht mehr die Konstante `KolEvent.close = "close"` nutzen? 

Dies ist ein klassischer Trade-off zwischen:

- ✅ Einfacherer Code (keine Imports, weniger Abhängigkeiten)
- ❌ Schlechtere Wartbarkeit (keine zentrale Definition, mehr Magic Strings)

Das wurde an vielen Stellen gemacht.

deleonio replied:
Der Test sollte nicht an die Impl gebunden sein.

laske185 reviewed (changes requested):
Wenn in den E2E-Tests weiterhin `KolEvents` genutzt wird, hat der PR kaum Änderungen.

deleonio commented on packages/components/src/utils/events.ts:
```diff
@@ -1,30 +1,34 @@
-		bubbles: true,
-		cancelable: true,
-		composed: true,
-		detail: detail,
```
Details fehlt.

deleonio replied:
Das `detail` Property ist **dynamisch** und variiert pro Event-Aufruf, während `DEFAULT_OPTIONS` nur die **statischen Event-Eigenschaften** enthält, die für alle KoliBri-Events gleich sind. Dies folgt der W3C CustomEvent API und stellt TypeScript-Typ-Sicherheit sicher.

deleonio commented on packages/components/src/components/single-select/shadow.tsx:
```diff
@@ -475,8 +484,14 @@ export class KolSingleSelect implements SingleSelectAPI {
 			case 'Enter': {
-				this.toggleListbox(event);
-				this._isOpen = false;
+				if (this._isOpen) {
```
Das auslagern, weil bei ' ' identisch oder zusammenlegen

deleonio committed:
refactor: rename kol* event names to native event names across components

- Updated event names in KolPopoverButton, KolPopover, KolSingleSelect, KolTableStateful, KolTableStateless, and KolTabs components to use standard DOM event names.
- Removed deprecated event names and adjusted related tests to reflect changes.
- Introduced a migration task to automate renaming of kol* events in the codebase.
- Added comprehensive tests to ensure correct renaming and functionality of events.

deleonio commented on packages/tools/kolibri-cli/test/rename-kol-event-names.spec.ts:
```diff
@@ -0,0 +1,437 @@
+export const noop = () => true;`;
+			fs.writeFileSync(tsPath, originalContent);
+
+			const task = RenameKolEventNamesTask.getInstance('^4');
+			task.run(tmpDir);
+
+			const content = fs.readFileSync(tsPath, 'utf8');
+			assert.strictEqual(content, originalContent);
+		});
+
+		it('handles empty files', () => {
+			const tsPath = path.join(tmpDir, 'empty.ts');
+			fs.writeFileSync(tsPath, '');
+
+			const task = RenameKolEventNamesTask.getInstance('^4');
+			task.run(tmpDir);
+
+			const content = fs.readFileSync(tsPath, 'utf8');
+			assert.strictEqual(content, '');
+		});
+
+		it('handles multiple occurrences of same event', () => {
+			const tsPath = path.join(tmpDir, 'multiple-same.ts');
+			fs.writeFileSync(
+				tsPath,
+				`element.addEventListener('kolChange', handler1);
+other.addEventListener('kolChange', handler2);
+third.addEventListener('kolChange', handler3);`,
+			);
+
+			const task = RenameKolEventNamesTask.getInstance('^4');
+			task.run(tmpDir);
+
+			const content = fs.readFileSync(tsPath, 'utf8');
+			const changeCount = (content.match(/addEventListener\('change'/g) || []).length;
+			assert.strictEqual(changeCount, 3);
+			assert.ok(!content.includes('kolChange'));
+		});
+	});
+});
```
@copilot open a new pull request to apply changes based on [this feedback](https://github.com/public-ui/kolibri/pull/9126#discussion_r2670740152)

Copilot committed:
Initial plan

deleonio committed:
refactor: remove unnecessary setTimeout calls in combobox, textarea, and renderer components

Copilot committed:
test: add comprehensive JSX prop renaming tests for event handlers

Co-authored-by: deleonio <6279703+deleonio@users.noreply.github.com>

deleonio committed:
test: add JSX/TSX event handler prop renaming tests (#9188)

laske185 commented on packages/components/src/components/form/form.e2e.ts:
```diff
@@ -31,20 +30,20 @@ test.describe('kol-form', () => {
-			['reset', KolEvent.reset],
+		const EVENTS: [string, string][] = [
+			['submit', 'submit'],
+			['reset', 'reset'],
```
Braucht man das jetzt noch? Also als Liste reicht auf jeden Fall.

laske185 replied:
Machen wir separat.

laske185 commented on packages/components/src/components/popover-button/shadow.tsx:
Warum wurden die Poperties entfernt?

deleonio replied:
Der Popover-Button hat "fälschlicherweise" die Props größtenteils vom Button geerbt. Viele davon machen aber keinen Sinn.
